en/setup/vscode.md

```diff
@@ -0,0 +1,55 @@
+# Visual Studio Code IDE
+
+Visual Studio Code (VSCode) is a powerful cross-platform source code editor/IDE that can be used for PX4 development on Linux and macOS (Windows support coming soon).
+
+PX4 has been set up with all the definition files that you need to install needed VSCode extensions, and and start developing.
+This topic explains how to get started.
+
+<!-- 
+- why VSCode vs Qt creator? 
+- Do we recommend VSCode?
+- what had to happen to make this setup work - ie presumably the directory contains definition files that vscode needs to know to set everything up?
+-->
+
+
+## Preconditions
+
+You must already have installed the command line [PX4 developer environment](../setup/dev_env.md) for your platform and downloaded the *Firmware* source code repo.
+
+<!-- xcode? -->
+
+## Installation & Setup
+
+1. [Download and install VSCode](https://code.visualstudio.com/) (you will be offered the correct version for your OS).
+1. Add the PX4 source code into VSCode. 
+   - Select the menu option **File > Open Folder** (or **File > Add Folder to Workspace**):
+     ![Menu - Open Folder/Add Folder to Workspace](../../assets/vscode/menu_file_open_folder.jpg)
+   - A file selection dialog will appear. 
+     Select the PX4 project directory (**Firmware**) from the file selection dialog to load the source.
+1. Press **Install All** on the *This workspace has extension recommendations* prompt (this will appear after a moment on the bottom right of the IDE).
+   ![Install extensions](../../assets/vscode/prompt_install_extensions.jpg)
+
+   VSCode will open the Extensions panel on the left hand side so you can watch the progress of installation.
+   ![PX4 loaded into VSCode Explorer](../../assets/vscode/installing_extensions.jpg)
+1. A number of notifications/prompts may appear in the bottom right corner (if these disappear, click the little "alarm" icon on the right of the blue bar at the bottom).
+   - If asked to install a new version of cmake you can say "no" on Ubuntu 1.18.04
+   - If prompted sign into github.com, do this if you wish to make pull requests directly from the IDE.
```

Comment:
Should we be recommending that they do this? 
Am I correct about what this does?

Reply:
Yes, although it can do everything, including reviewing pull requests with the diff shown in your local editor. I wouldn't necessarily recommend it one way or the other, it's just an optional thing that some people might like, and others probably won't.